src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -197,15 +204,15 @@ bool ScanAssembly(Assembly assembly, Dictionary<string, bool> processed)
 
             if (assembly.GetName().Name == CoreAssemblyName)
             {
-                processed[assembly.FullName] = true;
+                return processed[assembly.FullName] = false;
             }
 
             if (ShouldScanDependencies(assembly))
             {
                 foreach (var referencedAssemblyName in assembly.GetReferencedAssemblies())
                 {
                     var referencedAssembly = GetReferencedAssembly(referencedAssemblyName);
-                    var referencesCore = ScanAssembly(referencedAssembly, processed);
+                    var referencesCore = referencedAssembly.GetName().Name == CoreAssemblyName || ScanAssembly(referencedAssembly, processed);
```

Comment:
Why would this be needed? Why would we add Core as it would already be included.

Reply:
it's not adding core, we're checking whether the assembly references core, and if it does, we add this assembly (not core) to the "to scan" list.